src/lib.rs

```diff
@@ -731,16 +733,18 @@ impl<T: AsSocket> Async<T> {
     /// # std::io::Result::Ok(()) });
     /// ```
     pub fn new(io: T) -> io::Result<Async<T>> {
-        let borrowed = io.as_socket();
-
         // Put the socket in non-blocking mode.
-        set_nonblocking(borrowed)?;
+        set_nonblocking(io.as_socket())?;
+
+        Self::new_nonblocking(io)
+    }
 
+    fn new_nonblocking(io: T) -> io::Result<Async<T>> {
```

Comment:
Can we make this public to solve https://github.com/smol-rs/async-io/issues/132?

Reply:
Done in https://github.com/smol-rs/async-io/pull/159/commits/de07b6adca6432e2fde5ce2a15c66c41731d7992.